src/kibana/components/courier/data_source/_abstract.js

```diff
@@ -295,6 +295,23 @@ define(function (require) {
           };
 
           /**
+           * Evict Filters that doesn't match the index patterns
+           * @param {object} filter The filter to clean
+           * @returns {object}
+           */
+          var keepApplicableFilters = function (filter) {
+            if (!!(filter.index)) {
+              return filter.index === this.id;
+            } else if (!!(filter.meta) && !!(filter.meta.index)) {
+              return filter.meta.index === this.id;
+            } else if (!!(filter.geo_bounding_box)) {
+              var field = _.keys(filter.geo_bounding_box)[0];
```

Comment:
This filtering is required because otherwise defining geo-bounding boxes on unknown fields raises query errors. 

I tried to make this filtering as generic as possible so it applies to other kind of filters as well. Indeed refining all the queries of a dashboard with a drill down that exists only on some of the queries leads to empty widgets (as the filter won't match)

I agree you might want to make this a config option at the dashboard level, although I don't see any mechanism to do this in the current state of the code.